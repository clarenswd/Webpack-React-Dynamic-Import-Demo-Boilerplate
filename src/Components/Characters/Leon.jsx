import React from 'react';
import Loadable from 'react-loadable';
import LoadingMessage from 'Components/Common/LoadingMessage';
import InventoryWindow from 'Components/Windows/InventoryWindow';
import Knife from 'Components/Weapons/Knife';
// import Handgun from 'Components/Weapons/Handgun';
// import Magnum from 'Components/Weapons/Magnum';

const Handgun = Loadable({
  loader: () => import('Components/Weapons/Handgun' /* webpackChunkName: 'Handgun' */),
  loading: LoadingMessage,
});

const Magnum = Loadable({
  loader: () => import('Components/Weapons/Magnum' /* webpackChunkName: 'Magnum' */),
  loading: LoadingMessage,
});

const Leon = () => {
  const name = 'Leon Scott Kennedy';

  return (
    <div>
      <h2>{name}</h2>
      <InventoryWindow>
        <Knife />
        <Handgun />
        <Magnum />
      </InventoryWindow>
    </div>
  );
};

export default Leon;
