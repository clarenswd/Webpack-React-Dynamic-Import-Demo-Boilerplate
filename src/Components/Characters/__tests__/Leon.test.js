import React from 'react';
import renderer from 'react-test-renderer';
import Leon from '../Leon';

it('renders correctly', () => {
  const tree = renderer
    .create(<Leon />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});