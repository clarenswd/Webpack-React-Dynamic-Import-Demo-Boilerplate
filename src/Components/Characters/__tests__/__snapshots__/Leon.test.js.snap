// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Leon Scott Kennedy
  </h2>
  <div
    className="container"
  >
    <div
      className="knife"
    >
      Knife
    </div>
    <div
      className="loadingMessage"
    >
      Loading...
    </div>
    <div
      className="loadingMessage"
    >
      Loading...
    </div>
  </div>
</div>
`;
